File: scaffold/src/main/java/top/xuqingquan/web/x5/IndicatorController.java
package top.xuqingquan.web.x5;

import com.tencent.smtt.sdk.WebView;
import top.xuqingquan.web.agent.BaseIndicatorSpec;

public interface IndicatorController extends top.xuqingquan.web.agent.IndicatorController {

    void progress(WebView v, int newProgress);
}

File: scaffold/src/main/java/top/xuqingquan/webbak/publics/IndicatorController.kt
package top.xuqingquan.webbak.publics

import top.xuqingquan.webbak.nokernel.BaseIndicatorSpec

interface IndicatorController {

    fun progress(v: android.webkit.WebView, newProgress: Int)

    fun progress(v: com.tencent.smtt.sdk.WebView, newProgress: Int)

    fun offerIndicator(): BaseIndicatorSpec?

    fun showIndicator()

    fun setProgress(newProgress: Int)

    fun finish()
}
